### ISSUE TITLE
Translations

### ISSUE THREAD

**Issue by robdeemer** [davemacdo/composer-diversity#30]
We should come up with translated versions of both pages in a number of languages…if we're being diverse, might as well do it as much as possible. Of course, _we_ wouldn't do the translations, but we can find folks willing to help. Here's my off-the-cuff ideas:

- Spanish
- French
- German
- Portuguese
- Russian
- Chinese
- Japanese
- Korean
- Arabic

**Comment by davemacdo** [comment]
Yes. I think localization should be a part of the next step. There aren't that many strings we would need to translate. The only thing that could be tricky is translating country names. I'll have to see if I can find some off-the-shelf library that handles this. Otherwise, I'd be ok punting on that and using English country names and focusing on localizing the functional parts, if that's alright with you.

**Comment by robdeemer** [comment]
All sounds good...thanks!

Rob

————————
Dr. Rob Deemer
Head of Music Composition & Assistant to the President
American Council on Education Fellow (2016-2017)
School of Music, State University of New York at Fredonia

Chair, NYSSMA Composition & Improvisation Committee
Chair-Elect, NAfME Council of Music Composition

Composer-in-Residence, Buffalo Chamber Players
Composer-in-Residence, Harmonia Chamber Singers

3162 Mason Hall • Fredonia, NY 14063 • (716) 673-3133
e-mail: deemer@fredonia.edu • web: www.robdeemer.com

**Comment by ianring** [comment]
Hey I've figured out how to do this with "vue-i18n". 
http://kazupon.github.io/vue-i18n/

I've forked the repo, and I'll submit a pull request when it's all working. For now I'll only have an english lexicon and a 2nd dummy language for testing. Dave, you're right there aren't very many strings to translate.

In the first PR I'll only get the i18n working using a default language, and nothing will appear to have changed. The next step is to provide the user with a way to choose their language, and persist it in a cookie or something.

**Comment by ianring** [comment]
see https://github.com/davemacdo/composer-diversity/pull/34

**Comment by davemacdo** [comment]
I'm done poking at the i18n stuff for now, and I've fixed my terrible indentation practices. 

Regarding the UI, can we get the user's system language from the request headers? I've never tried.

**Comment by davemacdo** [comment]
@robdeemer Do you have anybody in mind who would be willing to do any of the translating? The list of text strings we need to translate for the main app is pretty small at the moment. I think this file should be somewhat self-explanatory, even if the formatting is a bit foreign. 

https://github.com/davemacdo/composer-diversity/blob/master/src/localized-strings.json

Lines 4 to 57 are all the blurbs in English. Lines 62 to 115 are a placeholder language with a random set of letters. We would only need those tiny handful of words translated, and since so many are common music terms, we could probably get most of them done ourselves (ie. _streichquartett_ or _pianoforte_). It's the more general things like "About the project" that we'd need help with. 

@ianring if you want to work on the implementation (could be as simple as [this](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorLanguage/language)), I could probably start the German, Italian, French, and Spanish localizations with standard music terms and leave the unknowns in English temporarily.

**Comment by robdeemer** [comment]
Good question. Let me see who I can scare up :)

————————
Dr. Rob Deemer
Head of Music Composition & Assistant to the President
American Council on Education Fellow (2016-2017)
School of Music, State University of New York at Fredonia

Chair, NYSSMA Composition & Improvisation Committee
Chair-Elect, NAfME Council of Music Composition

Composer-in-Residence, Buffalo Chamber Players
Composer-in-Residence, Harmonia Chamber Singers

3162 Mason Hall • Fredonia, NY 14063 • (716) 673-3133
e-mail: deemer@fredonia.edu • web: www.robdeemer.com

**Comment by robdeemer** [comment]
I’m on my phone...can you send me the list of languages I’ve suggested? Thanks!

Rob

————————
Dr. Rob Deemer
Head of Music Composition & Assistant to the President
American Council on Education Fellow (2016-2017)
School of Music, State University of New York at Fredonia

Chair, NYSSMA Composition & Improvisation Committee
Chair-Elect, NAfME Council of Music Composition

Composer-in-Residence, Buffalo Chamber Players
Composer-in-Residence, Harmonia Chamber Singers

3162 Mason Hall • Fredonia, NY 14063 • (716) 673-3133
e-mail: deemer@fredonia.edu • web: www.robdeemer.com

**Comment by ianring** [comment]
HTTP requests originating from a browser usually have a optional "Accept-Language" header. For example, mine is set to "en-US". And I think all modern browsers make it available in the window.navigator object.

var language = window.navigator.userLanguage || window.navigator.language;
alert(language);

IMHO we can discard variants like "en-CA" and "en-GB", and just use the major language code; ie we discard anything after a hyphen.

We can use the browser's language choice as a default, but I would be keen to let the user choose something else and store that in a cookie, if they want to.

**Comment by davemacdo** [comment]
That works! 

I agree that we can discard the variants for English, and probably other languages as well. Can I assign the implementation to you? I'm working with Rob to get the actual translations, and I can put those into localized-strings.json as we get them. I'll use the standard two-letter language codes. Right now, we have a volunteer for es, and we're looking for de, fr, and it to start.

**Comment by ianring** [comment]
Yes assign it to me

**Comment by davemacdo** [comment]
Huh. Apparently I can't Github-Assign you (which makes no sense, as I _can_ assign Rob, who has the same access that you do). But whatever, do your thing.

**Comment by ianring** [comment]
how odd. I'd like to figure out why, if for no other reason to have a better grasp of github. Do I need to add myself to a group, or a team, ... ? I don't see anything obvious on my end but I'll do a little research between cycles

**Comment by davemacdo** [comment]
https://github.com/isaacs/github/issues/100

That thread seems to indicate that it's not possible, but it doesn't explain why I could assign Rob, unless it's because he started the Issue.

**Comment by davemacdo** [comment]
Yep. Dumb as it may seem, only people have push access to the repo _or_ are the originator of an Issue can be assigned. How silly.

**Comment by ianring** [comment]
By the way, the time I can spend on this project will be enthusiastic but sporadic. So if anyone meanwhile wants to work on i18n stuff, don't hesitate on my account just because it's been sort-of assigned. Just drop a note here so we don't both write conflicting/overlapping solutions. Cheers ~
